sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeTableCommand.scala

```diff
@@ -88,14 +90,30 @@ case class AnalyzeTableCommand(tableName: String) extends RunnableCommand {
             }
           }.getOrElse(0L)
 
-        // Update the Hive metastore if the total size of the table is different than the size
+        val needUpdate = new mutable.HashMap[String, String]()
+        if (newTotalSize > 0 && newTotalSize != oldTotalSize) {
+          needUpdate += (AnalyzeTableCommand.TOTAL_SIZE_FIELD -> newTotalSize.toString)
+        }
+        if (!noscan) {
+          val oldRowCount = tableParameters.get(AnalyzeTableCommand.ROW_COUNT).map(_.toLong)
+            .getOrElse(-1L)
+          val newRowCount = sparkSession.table(tableName).count()
+
+          if (newRowCount >= 0 && newRowCount != oldRowCount) {
+            needUpdate += (AnalyzeTableCommand.ROW_COUNT -> newRowCount.toString)
+          }
+        }
+        // Update the Hive metastore if the above parameters of the table is different than those
         // recorded in the Hive metastore.
         // This logic is based on org.apache.hadoop.hive.ql.exec.StatsTask.aggregateStats().
-        if (newTotalSize > 0 && newTotalSize != oldTotalSize) {
+        if (needUpdate.nonEmpty) {
+          // need to set this parameter so that we can store other parameters like "numRows" into
+          // Hive metastore
+          needUpdate.put(AnalyzeTableCommand.STATS_GENERATED_VIA_STATS_TASK,
```

Comment:
If we don't set this parameter, "numRows" can not be stored into Hive metastore. We need to do this in Spark so that we can persist our statistics in metastore.